### PR TITLE
[RAC] Add `shouldCloseOnSelect` prop to DatePicker (type & docs only)

### PR DESCRIPTION
**Pull request by sookmax**
This PR doesn't close any existing issue.

Hello, I've noticed `DatePicker` component passes its props to `useDatePickerState` unmodified internally, which means I could use `shouldCloseOnSelect` feature if I wanted to in the current version already.

So, only things that are missing for proper usage of `shouldCloseOnSelect` on `DatePicker` are documentation and type definition, which this PR addresses.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### Documentation
1. yarn start:docs
2. head to http://localhost:1234/react-aria/DatePicker.html#datepicker-1
3. check if `shouldCloseOnSelect` is on the prop table.

### Types
1. Open packages/react-aria-components/stories/index.stories.tsx
2. Find `DatePickerExample`
3. Type `should` on `<DatePicker />` component and see if type IntelliSense works.
4. Specify `shouldCloseOnSelect={false}` on `<DatePicker />`
5. Visit http://localhost:9003/?path=/story/react-aria-components--date-picker-example&providerSwitcher-express=false&strict=true
6. Check if `shouldCloseOnSelect` works properly.

### CODE REVIEW

**Comment by snowystinger** [comment]
GET_BUILD

**Review by snowystinger — DISMISSED** [review]
I think this makes sense and it would be good to expose. Thanks for updating the DocsPackager as well, that was well done.
I've added a couple tests as well to validate our support for this prop.
Thanks again!

**Comment by reidbarber** [comment]
Went ahead and added this to DateRangePicker for parity.

File: packages/react-aria-components/test/DateRangePicker.test.js [hunk 1]
```diff
@@ -242,4 +242,39 @@ describe('DateRangePicker', () => {
...
+
+    await user.click(button);
+
+    let dialog = getByRole('dialog');
+
+    let cells = getAllByRole('gridcell');
+    let selected = cells.find(cell => cell.getAttribute('aria-selected') === 'true');
+    expect(selected.children[0]).toHaveAttribute('aria-label', 'Selected Range: Tuesday, January 10 to Friday, January 20, 2023, Tuesday, January 10, 2023 selected');
+
+    await user.click(selected.nextSibling.children[0]);
```
[the remark this answers is not in the document]
Comment (reidbarber) [reply packages/react-aria-components/test/DateRangePicker.test.js:277]: ahhh thanks

**Review by snowystinger — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]

**Pull request merged**